pkg/service/service.go

```diff
@@ -529,6 +535,15 @@ func (s *Service) createSVCInfoIfNotExist(p *lb.SVC) (*svcInfo, bool, bool,
 		s.svcByID[p.Frontend.ID] = svc
 		s.svcByHash[hash] = svc
 	} else {
+		// Reviewers: Local Redirect Policies with service matcher would have same frontend
+		// as the service clusterIP type. In such cases, if a Local redirect service
+		// exists, we shouldn't override it with clusterIP type (e.g., k8s event/sync, etc).
+		if svc.svcType == lb.SVCTypeLocalRedirect && p.Type == lb.SVCTypeClusterIP {
+			err := fmt.Errorf("Local redirect service exists for "+
+				"frontend %v, skip update for svc %s", p.Frontend, p.Name)
+			return svc, !found, prevSessionAffinity, prevLoadBalancerSourceRanges, err
```

Comment:
(For a lack of appropriate place to discuss, related to this case but not specifically this line..)

I think this kind of logic may  also need to be taken into account on startup when handling `Service.SyncWithK8sFinished()` here in `pkg/service/service.go`. I saw this message in my Cilium logs on startup after having the LRP from the GSG installed:

```
level=warning msg="Deleting no longer present service" l3n4Addr="{IP:169.254.169.254 L4Addr:{Protocol:NONE Port:8080} Scope:0}" serviceID=5 subsys=service
```

I think that's basically saying that Cilium started up and tried to sync the LB maps from BPF with the kubernetes services, and found that this service is missing. However this is likely because that logic is not taking LRP into account.

For testing this kind of case, I suspect that having LRP + pod + some constant connectivity check going through while restarting Cilium would likely help. Ideally there should be no drop on the LRP assuming the backend is still there and so on.

I think for a first ("beta") cut of the LRP feature this would be fine as-is (ie merge without necessarily addressing), but this is definitely indicative of a bug in synchronizing services on startup related to this feature. If this is not addressed in the initial PR, please be sure to add this to the list of followups or file an issue for it.